### DISCUSSION TITLE
error cannot download many videos

### DISCUSSION THREAD

**Discussion by thanhvt1509** [JoeanAmier/XHS-Downloader#74] [General]
Hello, I think your project is very cool. During use, I saw an error that this program could not download a certain number of videos, then it would error "Server disconnected.During handling of the above exception, another exception occurred" and " cannot access local variable ' temp' where it is not associated with a value ". If you can read it, please help me answer it

**Comment by JoeanAmier** [comment]
Please provide a screenshot.

**Reply by thanhvt1509** [reply]
![image](https://github.com/JoeanAmier/XHS-Downloader/assets/92007722/46822c93-b86f-4339-af7a-40f8bfe257a4)
![image](https://github.com/JoeanAmier/XHS-Downloader/assets/92007722/e2139b41-53a8-4cbf-865f-882209d6c962)
![image](https://github.com/JoeanAmier/XHS-Downloader/assets/92007722/a9beecc6-331e-427c-96d2-d17efab1a291)
![image](https://github.com/JoeanAmier/XHS-Downloader/assets/92007722/ee54f165-0b5c-47c6-9402-b0f4eeacabd8)
![image](https://github.com/JoeanAmier/XHS-Downloader/assets/92007722/4458f8de-78dd-47cf-b80d-60558a7e0bdf)
![image](https://github.com/JoeanAmier/XHS-Downloader/assets/92007722/08ee0164-23be-42df-bf90-7668e73040f6)
![image](https://github.com/JoeanAmier/XHS-Downloader/assets/92007722/2a86b3a0-e7df-4147-8054-8188d931ecdf)
![image](https://github.com/JoeanAmier/XHS-Downloader/assets/92007722/5facf4f1-754e-4966-b211-a02b226774b9)
![image](https://github.com/JoeanAmier/XHS-Downloader/assets/92007722/f182cd7c-807f-4dc9-b6bf-d32f29825f0b)
help me please

**Comment by JoeanAmier** [comment]
Is it the latest source code? What operation was in progress before the error occurred.

**Reply by thanhvt1509** [reply]
It is the latest source code. Before the error occurred, I was downloading all videos of 1 xhs channel

**Reply by JoeanAmier** [reply]
Provide links that cause errors.

**Reply by thanhvt1509** [reply]
I used the user script you provided to batch extract links however when running the program I encountered the above core error. When I try to run the program again, I still get the same error at some links. Do you have any way to fix this situation?

**Reply by JoeanAmier** [reply]
I was unable to reproduce your mistake, Can you tell me the complete operation steps?

**Reply by thanhvt1509** [reply]
https://youtu.be/8Rtibr4R3Pw 
After I used the user command file you provided to extract a row of links from an xhs user, the program ran with an error like the video. I have a suspicion that maybe my error is due to a network problem, but I have increased the delay time but it still happens.

**Reply by JoeanAmier** [reply]
Thank you for your feedback. The issue has been confirmed and will be fixed in the future.

**Reply by BiaBia9999** [reply]
@thanhvt1509 : anh đã fix được lỗi down multi link bị disconnect rất nhiều chưa anh, em dùng bị như vậy

**Reply by BiaBia9999** [reply]
hỏi ko thấy trả lời luôn

**Comment by JoeanAmier** [comment]
If you can run the program through source code, you can try the latest `2.0 Beta` version.

**Reply by thanhvt1509** [reply]
I used your latest source code, it no longer had the error I had before. Thank you for accepting my feedback. Have a nice day, hope you can develop the program in the near future

**Reply by BiaBia9999** [reply]
i cant download many links video xiaohongshu at the same time  ! can u fix it in next version 2.0 ??
It often disconnects

**Reply by JoeanAmier** [reply]
Please create a new issue or discussion feedback exceptions.
